client/session.go

```diff
@@ -1012,16 +1042,22 @@ func (s *session) fetchAllAttempt(
 			if atomic.AddInt32(&resultsAccessors, -1) == 0 {
 				s.iteratorArrayPool.Put(results)
 			}
+			if atomic.AddInt32(&idAccessors, -1) == 0 {
+				tsID.Finalize()
```

Comment:
Hm just trying to remember why we duplicate the finalization in both completion callbacks?